internal/engine/compiler/impl_amd64_test.go

```diff
@@ -441,6 +441,61 @@ func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
 	}
 }
 
+// Relates to #1111: older AMD64 CPUs do not support the LZCNT instruction
+// CPUID should be used instead. We simulate presence/absence of the feature
+// by overriding the field in the corresponding struct.
+func TestAmd64Compiler_ensureClz_ABM(t *testing.T) {
+	tests := []struct {
+		name        string
+		cpuFeatures cpuFeatureFlags
+		expectedLen int
+	}{
+		{
+			name:        "with ABM",
+			expectedLen: 140,
+			cpuFeatures: cpuFeatureFlags{
+				flags:      0,
+				extraFlags: cpuExtraFeatureABM,
+			},
+		},
+		{
+			name:        "without ABM",
+			expectedLen: 156,
+			cpuFeatures: cpuFeatureFlags{
+				flags:      0,
+				extraFlags: 0, // no flags, thus no ABM, i.e. no LZCNT
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			env := newCompilerEnvironment()
+
+			newCompiler := func() compiler {
+				c := newCompiler().(*amd64Compiler)
+				// override auto-detected CPU features with the test case
+				c.cpuFeatures = tt.cpuFeatures
+				return c
+			}
+
+			compiler := env.requireNewCompiler(t, newCompiler, nil)
+			err := compiler.compilePreamble()
+			require.NoError(t, err)
+
+			err = compiler.compileClz(&wazeroir.OperationClz{Type: wazeroir.UnsignedInt64})
+			require.NoError(t, err)
+			//
+			err = compiler.compileReturnFunction()
+			require.NoError(t, err)
```

Comment:
same

Reply:
without this (only `without_ABM`)

```
    require.go:312: expected no error, but was jump target must not be nil for relative JMP: JMP {<nil>}
        /Users/evacchi/Devel/github.com/tetratelabs/wazero/internal/engine/compiler/impl_amd64_test.go:492
```

Reply:
the jump that is generated in "without_ABM" needs a "next" instruction so, I am just adding a NOP